apps/provisioning/pkg/apis/provisioning/v0alpha1/connections.go

```diff
@@ -0,0 +1,96 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+)
+
+// When this code is changed, make sure to update the code generation.
+// As of writing, this can be done via the hack dir in the root of the repo: ./hack/update-codegen.sh provisioning
+// If you've opened the generated files in this dir at some point in VSCode, you may also have to re-open them to clear errors.
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Connection struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ConnectionSpec   `json:"spec,omitempty"`
+	Secure ConnectionSecure `json:"secure,omitzero,omitempty"`
+	Status ConnectionStatus `json:"status,omitempty"`
+}
+
+type ConnectionSecure struct {
+	// Reference to the private key for GitHub App authentication
+	// This value is stored securely and cannot be read back
+	PrivateKey common.InlineSecureValue `json:"privateKey,omitzero,omitempty"`
```

Comment:
To make it generic for the Gitlab and Bitbucket (they don't use private keys).
```suggestion
	Secret common.InlineSecureValue `json:"secret,omitzero,omitempty"`
```

Or we will settle on different fields: 

```suggestion
	PrivateKey common.InlineSecureValue `json:"privateKey,omitzero,omitempty"`
	ClientSecret common.InlineSecureValue `json:"clientSecret,omitzero,omitempty"`
```

Probably the option 2 is the best for clarity.